### ISSUE TITLE
Google support broken

### ISSUE THREAD

**Issue by CorentinDoue** [floydspace/serverless-esbuild#105]
## Description
Using `serverless-esbuild >= v1.7.0` with `google` provider cause an error
```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
      at validateString (internal/validators.js:120:11)
      at Object.dirname (path.js:1128:5)
      at EsbuildPlugin.<anonymous> (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:141:223)
      at Generator.next (<anonymous>)
      at /home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:4:12)
      at /home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:140:76
      at Array.map (<anonymous>)
      at EsbuildPlugin.<anonymous> (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:140:51)
      at Generator.next (<anonymous>)
      at /home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:4:12)
      at EsbuildPlugin.bundle (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:137:16)
      at EsbuildPlugin.<anonymous> (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:72:28)
      at Generator.next (<anonymous>)
      at /home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:8:71
      at new Promise (<anonymous>)
      at __awaiter (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:4:12)
      at Object.before:package:createDeploymentArtifacts [as hook] (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless-esbuild/dist/index.js:71:63)
      at PluginManager.invoke (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless/lib/classes/PluginManager.js:551:20)
      at PluginManager.spawn (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless/lib/classes/PluginManager.js:573:5)
      at Object.before:deploy:deploy [as hook] (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless/lib/plugins/deploy.js:53:13)
      at PluginManager.invoke (/home/runner/work/monorepo/monorepo/serverless/set-order-datastore/node_modules/serverless/lib/classes/PluginManager.js:551:9)
```

## Cause

The commit https://github.com/floydspace/serverless-esbuild/commit/7325a7857babaf4d5b6a3c2f4692e115c330ef59 changes the return type of `src/helper.ts:extractFileNames` from `string[]` to `{ entry: string; func: any }[]` for all provider except `google`.

If the provider is `google`, the return type is still `string[]` and `src/index/ts:bundle` fails while iterating on the array: `rootFileNames.map(async ({ entry, func }) => {`

## Workaround

downgrade `serverless-esbuild` to` v1.6.0`

**Comment by olup** [comment]
Thanks for reporting this ! As it is mentioned somewhere google support is indeed a little experimental since those changes, as I could not test the provider properly. I will fix this asap (on holiday now so could be a couple of days)

**Comment by olup** [comment]
@CorentinDoue - I am reading gcp doc and see that we have two ways to handle the file naming - either by having an `index` or by specifying the main in `package.json`. just curious here, which one do you use ? Also, are you hoping to use individual packaging ?

**Comment by olup** [comment]
@CorentinDoue would you mind trying to use #107 to bundle your function on GCP, and tell me if this fixes your problem? Packaging works alright, however I could not deploy it on GCP to further check that function executes properly (no GCP project and account at hand ATM).

**Comment by CorentinDoue** [comment]
I tested it and it looks good. Thank you

**Comment by christopher-caldwell** [comment]
Does anyone have a guide or a link to get GCP running with this plugin? AWS works like a charm, but GCP is really confusing (for me). I do understand support for GCP is limited to experimental, but does anyone have a starting point to go from?

**Comment by floydspace** [comment]
Hi @christopher-caldwell , I don't expect it's working for GCP to be honest. It's never been tested with gcp by me personally. Originally the plug-in was forked from 'serverless-typescript-plugin' which was supporting gcp, but afterward a lot of new features were implemented without taking into account and testing against gcp. So I would not expect much here.
But contributions are welcome, unless they do not break the current implementation.

**Comment by christopher-caldwell** [comment]
@floydspace I've been doing some digging and I think I found why this isn't working with GCP. 

In [this function](https://github.com/floydspace/serverless-esbuild/blob/0ec2ab8bab2443e0033a654daa3c8baf08058680/src/index.ts#L132), it's checking to see if it's node by comparing the _**AWS**_ runtimes which are not the same strings as the GCP plugin. This leads to them all being ignored.

**Comment by christopher-caldwell** [comment]
Is there a need to check for `aws` first? Theoretically, you could just check the runtimes in a flattened map.

```ts
{
    'nodejs16.x': 'node16',
    'nodejs14.x': 'node14',
    'nodejs12.x': 'node12',
    'nodejs16': 'node16',
    'nodejs14': 'node14',
    'nodejs12': 'node12',
}
```

**Comment by descampsk** [comment]
I make a PR to fix this #429
